src/components/external-source/ExternalSourceUpdateCard.svelte

```diff
@@ -0,0 +1,159 @@
+
+  $: sources.forEach(source => {
+    let sourceType = source.source_type_name;
+    let derivationGroup = source.derivation_group_name;
+    if (sourceType && derivationGroup) {
+      if (mappedSources[sourceType] && mappedSources[sourceType][derivationGroup]) {
+        mappedSources[sourceType][derivationGroup]?.push(source);
+      } else if (mappedSources[sourceType] && !mappedSources[sourceType][derivationGroup]) {
+        mappedSources[sourceType][derivationGroup] = [source];
+      } else if (!mappedSources[sourceType]) {
+        mappedSources[sourceType] = { [derivationGroup]: [source] };
+      }
+    }
+  });
+</script>
+
+<div
+  class="card st-typography-label"
+  style="--border-color: {deleted ? 'rgba(152, 35, 35, 0.5)' : 'rgba(152, 101, 35, 0.5)'}"
+>
+  <div
+    class="card--row card--title-row"
+    style=" background: {deleted ? 'rgb(254, 234, 234)' : 'rgb(254, 252, 234)'};display:flex"
+  >
+    <div
+      class="card--title t-typography-medium"
+      style=" align-items: center; display: flex; gap: 5px;padding-left:5px; width: 100%"
+    >
+      <LightningCharge
+        class="filter-search-icon"
+        style="color: {deleted ? 'rgba(152, 35, 35, 0.5)' : 'rgba(152, 101, 35, 0.5)'}"
+      />
+      {#if sources.length === 1 && !deleted}
+        1 new file has been uploaded
+      {:else if sources.length === 1 && deleted}
+        1 file has been deleted
+      {:else if deleted}
+        {sources.length} files have been deleted
+      {:else}
+        {sources.length} new files have been uploaded
+      {/if}
+    </div>
+    <slot name="right">
+      <button class="st-button icon fs-6" on:click={() => dispatch('dismiss')}>
+        <XIcon />
+      </button>
+    </slot>
+  </div>
+  <div style="padding: 4px 14px 12px;">
+    <div class="card--row" style="padding-top: 5px">
+      <span class="st-typography-body">
+        {!deleted
+          ? 'New files matching source types and derivation groups in the current plan:'
+          : 'Deleted files organized by source type and derivation group:'}
+      </span>
+    </div>
+    <div class="card--body st-typography-body">
+      {#each Object.keys(mappedSources) as sourceType}
+        <Collapse title={sourceType} tooltipContent={'Source Type'} defaultExpanded={true}>
+          {#each Object.keys(mappedSources[sourceType]) as derivationGroup}
+            <Collapse title={derivationGroup} tooltipContent={'Derivation Group'} defaultExpanded={true}>
+              {#each mappedSources[sourceType][derivationGroup] as source}
+                <div class="card--source-info">
+                  <p
+                    style="flex-shrink: 1; max-width: 100%; overflow: hidden; text-overflow: ellipsis; white-space: nowrap"
+                  >
+                    {source.key}
+                  </p>
+                  <p
+                    style=" color: gray;flex-shrink: 0; margin-left: auto; padding-right: 0.25rem; padding-top: 0.1rem; white-space: nowrap"
```

Comment:
Could you remove the padding on this? Causes the time ago to not be aligned with the source key text.